onnx/shape_inference/implementation.cc

```diff
@@ -146,6 +142,8 @@ void mergeShapesAndTypes(
 void mergeShapesAndTypes(
     const TypeProto& inferredType,
     TypeProto* existingType) {
+  // Check before merge
+  checkShapesAndTypes(inferredType, *existingType);
   if (inferredType.has_tensor_type()) {
```

Comment:
We need to add a case for "if (inferredType.value_case() == VALUE_NOT_SET)".

Reply:
Sorry, it was a  copy-paste error. I meant the case existingType.value_case == VALUE_NOT_SET needs to be handled. We need to copy the type from inferredType to existingType.

Reply:
Ok, please ignore my comment. I think the following code works fine. "existingType->mutable_tensor_type()" will create the tensor_type if not already present.